### PR TITLE
Fix multidomain AutomaticMechanismAnalysis

### PR DESCRIPTION
**Pull request by hwpang**
Bigger changes:

1. Append the global p vector (corresponding to multidomain) to each Simulation object in SystemSimulation, instead of the local p vector (corresponding to one domain). Append interfaces as well.
2. Uses the global p vector to calculate dSdt and the corresponding global indexes for species and reactions

### CODE REVIEW

**Comment by hwpang** [comment]
Previous push crashes single domain AMA. New push fixes it.

**Comment by hwpang** [comment]
Previous push crashes the test due to incorrect indexing. New push fixes it.

**Comment by hwpang** [comment]
New push fixes a bug related to the test for multidomain AMA

**Comment by hwpang** [comment]
Previous push encountered package version incompatibility problem after removing and reinstalling numpy, as well as installing nomkl. New push removed the related code.

File: src/AutomaticMechanismAnalysis.jl [hunk 1]
```diff
@@ -280,7 +280,7 @@ function analyzespc(sim,spcname,t;N=10,tol=1e-3,branchthreshold=0.9,
     for rxnind in sensinds
         branches,rps = getbranchpathinfo(sim,spcind,rxnind,ropp,ropl,rts;steptol=steptol,branchtol=branchtol)
         radprodlossfract = getradprodlossfract(sim,rxnind,rts)
-        if eliminate
+        if eliminate && !isempty(rps)
```
Comment (mjohnson541) [inline src/AutomaticMechanismAnalysis.jl:283]: What if we want to eliminate unimportant branches even if we don't have any reaction paths?

File: src/Simulation.jl [hunk 2]
```diff
@@ -47,7 +47,7 @@ struct SystemSimulation{Q,B<:AbstractODESolution,X,Y,Z}
 end
 
 function SystemSimulation(sol,domains,interfaces,p)
-    sims = Tuple([Simulation(sol,domain) for domain in domains])
+    sims = Tuple([Simulation(sol,domain,interfaces,p) for domain in domains])
```
Comment (mjohnson541) [inline src/Simulation.jl:50]: Do we want each Simulation object to have all of the interface objects in the system?

File: deps/build.jl [hunk 3]
```diff
@@ -3,12 +3,12 @@ if PyCall.pyversion.major != 3 || PyCall.pyversion.minor != 7
...
-    catch e 
-    end
-    Conda.add("nomkl")
-    Conda.add("numpy")
+    # try
+    #     Conda.rm("numpy") #get around MKL problem
+    # catch e 
+    # end
+    # Conda.add("nomkl")
+    # Conda.add("numpy")
```
Comment (mjohnson541) [inline deps/build.jl:11]: Are we sure we want to do this on main?

**Review by mjohnson541 — COMMENTED** [review]
Largely looks good. Some comments and questions.

File: src/AutomaticMechanismAnalysis.jl (hunk 1 above)
Comment (hwpang) [reply src/AutomaticMechanismAnalysis.jl:283]: You're right, I will remove this

File: src/Simulation.jl (hunk 2 above)
Comment (hwpang) [reply src/Simulation.jl:50]: You're right, they probably only need to carry interfaces that are connected to them

File: deps/build.jl (hunk 3 above)
Comment (hwpang) [reply deps/build.jl:11]: Back then it was the only way I can get the CI test to pass for some reasons. I will remove them in the new push and see how it goes.

**Comment by hwpang** [comment]
@mjohnson541 I have addressed all the comments.

**Comment by hwpang** [comment]
@mjohnson541 I have addressed all your comments & all tests have passed.

**Pull request merged**